creation/lib/cgWParamDict.py

```diff
@@ -297,15 +297,58 @@ def populate(self, other=None):
         self.dicts["untar_cfg"].add(pychirp_tarball, "lib/python/htchirp")
 
         # Add cvmfsexec
-        cvmfsexec_tarball = "cvmfs_utils.tar.gz"
+        # Add cvmfsexec helper script enabled by conditional download
+        cvmfs_helper = "cvmfs_helper_funcs.sh"
         self.dicts["file_list"].add_from_file(
-            cvmfsexec_tarball,
+            cvmfs_helper,
             cWDictFile.FileDictFile.make_val_tuple(
-                cWConsts.insert_timestr(cvmfsexec_tarball), "untar", cond_download="GLIDEIN_USE_CVMFSEXEC"
+                cWConsts.insert_timestr(cvmfs_helper), "exec", cond_download="GLIDEIN_USE_CVMFSEXEC"
             ),
-            os.path.join(cgWConsts.WEB_BASE_DIR, cvmfsexec_tarball),
+            os.path.join(cgWConsts.WEB_BASE_DIR, cvmfs_helper),
         )
-        self.dicts["untar_cfg"].add(cvmfsexec_tarball, "cvmfs_utils")
+
+        ### for dynamic selection of cvmfsexec distribution -- block start
+        dist_select_script = "cvmfsexec_platform_select.sh"
+        self.dicts["file_list"].add_from_file(
+            dist_select_script,
+            cWDictFile.FileDictFile.make_val_tuple(
+                cWConsts.insert_timestr(dist_select_script), "exec", cond_download="GLIDEIN_USE_CVMFSEXEC"
+            ),
+            os.path.join(cgWConsts.WEB_BASE_DIR, dist_select_script),
+        )
+
+        # TODO: update the following line to use the temporary location of the tarballs
+        # TODO: currently uses the 'cvmfsexec' dir inside creation/web-base
+        # TODO: the 'cvmfsexec' dir has all the tarballs created as a result of running 'generate_cvmfsexec_distros.sh'
+        # get the location of the tarballs created during reconfig/upgrade
+        distros_loc = os.path.join(self.work_dir, "cvmfsexec/tarballs")
+        distros = os.listdir(distros_loc)
+        if len(distros) == 0:
+            print("Distributions for cvmfsexec not found... Skipping tarball creation.")
+        else:
+            for cvmfsexec_idx in range(len(distros)):  # TODO: os.scandir() is more efficient with python 3.x
+                distro_info = distros[cvmfsexec_idx].split("_")
+                distro_arch = (distro_info[3] + "_" + distro_info[4]).split(".")[0]
```

Comment:
enclose in try except for IndexOutOfRange, with a warning message (invalid subdirectory `cvmfsexec/tarballs{distro_info}`) and a continue

Reply:
this has been changed (see latest commits)